### PR TITLE
Dynamic Simulation - Curve configuration - Endpoints to evaluate a filter and to get all countries of network

### PR DESCRIPTION
**Pull request by thangqp**
Related PRs:
Front-end : 
https://github.com/gridsuite/gridstudy-app/pull/1658 (with demo)
Back-end :
https://github.com/gridsuite/network-map-server/pull/147
https://github.com/gridsuite/filter-server/pull/78

### CODE REVIEW

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 1]
```diff
@@ -467,6 +467,17 @@ public ResponseEntity<String> getNetworkElementInfos(
         return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(studyService.getNetworkElementInfos(studyUuid, nodeUuid, elementType, infoType, elementId, operation, inUpstreamBuiltParentNode));
     }
 
+    @GetMapping(value = "/studies/{studyUuid}/nodes/{nodeUuid}/network-map/countries")
+    @Operation(summary = "Get network countries")
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The list of countries")})
+    public ResponseEntity<String> getNetworkCountries(
+            @Parameter(description = "Study uuid") @PathVariable("studyUuid") UUID studyUuid,
+            @Parameter(description = "Node uuid") @PathVariable("nodeUuid") UUID nodeUuid,
+            @Parameter(description = "Should get in upstream built node ?") @RequestParam(value = "inUpstreamBuiltParentNode", required = false, defaultValue = "false") boolean inUpstreamBuiltParentNode) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/StudyController.java:476]: Add inUpstreamBuiltParentNode=true

File: src/main/resources/application-local.yml [hunk 2]
```diff
@@ -67,4 +67,7 @@ gridsuite:
     -
       name: voltage-init-server
       base-uri: http://localhost:5038
+    -
```
Comment (SlimaneAmar) [inline src/main/resources/application-local.yml:70]: Need to add in `application.yml` file

And `deployment/k8s/resources/study/config/study-server-application.yml`

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 3]
```diff
@@ -1761,4 +1772,16 @@ public ResponseEntity<AboutInfo[]> getSuiteAboutInformation(
...
                 remoteServicesInspector.convertServicesInfoToAboutInfo(Objects.requireNonNullElseGet(suiteServersInfo.getBody(), Map::of)));
     }
+
+    @PostMapping(value = "/studies/{studyUuid}/nodes/{nodeUuid}/filters/evaluate")
+    @Operation(summary = "Evaluate a filter to get matched elements")
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The list of matched elements")})
+    public ResponseEntity<String> evaluateFilter(
+            @Parameter(description = "Study uuid") @PathVariable("studyUuid") UUID studyUuid,
+            @Parameter(description = "Node uuid") @PathVariable("nodeUuid") UUID nodeUuid,
+            @Parameter(description = "Should get in upstream built node ?") @RequestParam(value = "inUpstreamBuiltParentNode", required = false, defaultValue = "false") boolean inUpstreamBuiltParentNode,
```
[the remark this answers is not in the document]
Comment (thangqp) [reply src/main/java/org/gridsuite/study/server/StudyController.java:1782]: Corrected in request from the front-end
https://github.com/gridsuite/gridstudy-app/pull/1658/commits/fef4bab75d83dc7ef8bd57d92ba9343d123b6c51#diff-004e2fd93b0e13e660ba95c75ebe656a464ba5f4810c28fd61ce1feb353ceaf6R24

File: src/main/java/org/gridsuite/study/server/StudyController.java (hunk 1 above)
Comment (thangqp) [reply src/main/java/org/gridsuite/study/server/StudyController.java:476]: Corrected in request from front-end
https://github.com/gridsuite/gridstudy-app/pull/1658/commits/fef4bab75d83dc7ef8bd57d92ba9343d123b6c51#diff-a05894790b3ecf0b35dce0445a1317e2172b6739be46d88b524702c5085fe60bR136

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**